From 1decf05d0f4de78ef67dc3f794709258c689e09e Mon Sep 17 00:00:00 2001
From: Emmanuel Grumbach <emmanuel.grumbach@intel.com>
Date: Tue, 18 Jun 2024 19:25:56 +0300
Subject: [PATCH] wifi: mac80211: inform the low level if drv_stop() is a
 suspend

This will allow the low level driver to take different actions for
different flows.

Signed-off-by: Emmanuel Grumbach <emmanuel.grumbach@intel.com>
Signed-off-by: Miri Korenblit <miriam.rachel.korenblit@intel.com>
Link: https://patch.msgid.link/20240618192529.739036208b6e.Ie18a2fe8e02bf2717549d39420b350cfdaf3d317@changeid
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 drivers/net/wireless/admtek/adm8211.c             |  2 +-
 drivers/net/wireless/ath/ar5523/ar5523.c          |  2 +-
 drivers/net/wireless/ath/ath10k/mac.c             |  2 +-
 drivers/net/wireless/ath/ath11k/mac.c             |  2 +-
 drivers/net/wireless/ath/ath12k/mac.c             |  2 +-
 drivers/net/wireless/ath/ath5k/base.c             |  2 +-
 drivers/net/wireless/ath/ath5k/base.h             |  2 +-
 drivers/net/wireless/ath/ath9k/htc_drv_main.c     |  2 +-
 drivers/net/wireless/ath/ath9k/main.c             |  2 +-
 drivers/net/wireless/ath/carl9170/main.c          |  2 +-
 drivers/net/wireless/ath/wcn36xx/main.c           |  2 +-
 drivers/net/wireless/atmel/at76c50x-usb.c         |  2 +-
 drivers/net/wireless/broadcom/b43/main.c          |  2 +-
 drivers/net/wireless/broadcom/b43legacy/main.c    |  2 +-
 .../broadcom/brcm80211/brcmsmac/mac80211_if.c     |  2 +-
 drivers/net/wireless/intel/iwlegacy/3945-mac.c    |  2 +-
 drivers/net/wireless/intel/iwlegacy/4965-mac.c    |  2 +-
 drivers/net/wireless/intel/iwlegacy/4965.h        |  2 +-
 drivers/net/wireless/intel/iwlwifi/dvm/mac80211.c |  2 +-
 drivers/net/wireless/intel/iwlwifi/mvm/mac80211.c |  2 +-
 drivers/net/wireless/intel/iwlwifi/mvm/mvm.h      |  2 +-
 drivers/net/wireless/intersil/p54/main.c          |  2 +-
 drivers/net/wireless/marvell/libertas_tf/main.c   |  2 +-
 drivers/net/wireless/marvell/mwl8k.c              |  4 ++--
 drivers/net/wireless/mediatek/mt76/mt7603/main.c  |  2 +-
 drivers/net/wireless/mediatek/mt76/mt7615/main.c  |  2 +-
 drivers/net/wireless/mediatek/mt76/mt7615/usb.c   |  2 +-
 drivers/net/wireless/mediatek/mt76/mt76x0/pci.c   |  2 +-
 drivers/net/wireless/mediatek/mt76/mt76x0/usb.c   |  2 +-
 .../net/wireless/mediatek/mt76/mt76x2/pci_main.c  |  2 +-
 .../net/wireless/mediatek/mt76/mt76x2/usb_main.c  |  2 +-
 drivers/net/wireless/mediatek/mt76/mt7915/main.c  |  2 +-
 drivers/net/wireless/mediatek/mt76/mt7921/main.c  |  4 ++--
 drivers/net/wireless/mediatek/mt76/mt792x.h       |  4 ++--
 drivers/net/wireless/mediatek/mt76/mt792x_core.c  |  2 +-
 drivers/net/wireless/mediatek/mt76/mt792x_usb.c   |  4 ++--
 drivers/net/wireless/mediatek/mt76/mt7996/main.c  |  2 +-
 drivers/net/wireless/mediatek/mt7601u/main.c      |  2 +-
 drivers/net/wireless/purelifi/plfxlc/mac.c        |  2 +-
 drivers/net/wireless/purelifi/plfxlc/mac.h        |  2 +-
 drivers/net/wireless/purelifi/plfxlc/usb.c        |  4 ++--
 drivers/net/wireless/ralink/rt2x00/rt2x00.h       |  2 +-
 drivers/net/wireless/ralink/rt2x00/rt2x00mac.c    |  2 +-
 .../net/wireless/realtek/rtl818x/rtl8180/dev.c    |  2 +-
 .../net/wireless/realtek/rtl818x/rtl8187/dev.c    |  2 +-
 drivers/net/wireless/realtek/rtl8xxxu/core.c      |  2 +-
 drivers/net/wireless/realtek/rtlwifi/core.c       |  4 ++--
 drivers/net/wireless/realtek/rtw88/mac80211.c     |  2 +-
 drivers/net/wireless/realtek/rtw89/mac80211.c     |  2 +-
 drivers/net/wireless/rsi/rsi_91x_mac80211.c       |  3 ++-
 drivers/net/wireless/silabs/wfx/sta.c             |  2 +-
 drivers/net/wireless/silabs/wfx/sta.h             |  2 +-
 drivers/net/wireless/st/cw1200/sta.c              |  2 +-
 drivers/net/wireless/st/cw1200/sta.h              |  2 +-
 drivers/net/wireless/ti/wl1251/main.c             |  2 +-
 drivers/net/wireless/ti/wlcore/main.c             |  2 +-
 drivers/net/wireless/virtual/mac80211_hwsim.c     |  2 +-
 drivers/net/wireless/zydas/zd1211rw/zd_mac.c      |  2 +-
 drivers/net/wireless/zydas/zd1211rw/zd_mac.h      |  2 +-
 drivers/net/wireless/zydas/zd1211rw/zd_usb.c      |  2 +-
 drivers/staging/vt6655/device_main.c              |  2 +-
 drivers/staging/vt6656/main_usb.c                 |  2 +-
 include/net/mac80211.h                            |  2 +-
 net/mac80211/driver-ops.c                         |  6 +++---
 net/mac80211/driver-ops.h                         |  2 +-
 net/mac80211/ieee80211_i.h                        |  2 +-
 net/mac80211/iface.c                              |  4 ++--
 net/mac80211/pm.c                                 |  4 ++--
 net/mac80211/trace.h                              | 15 ++++++++++++---
 net/mac80211/util.c                               |  4 ++--
 70 files changed, 93 insertions(+), 83 deletions(-)

--- a/mt7603/main.c
+++ b/mt7603/main.c
@@ -23,7 +23,7 @@ mt7603_start(struct ieee80211_hw *hw)
 }
 
 static void
-mt7603_stop(struct ieee80211_hw *hw)
+mt7603_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt7603_dev *dev = hw->priv;
 
--- a/mt7615/main.c
+++ b/mt7615/main.c
@@ -91,7 +91,7 @@ out:
 	return ret;
 }
 
-static void mt7615_stop(struct ieee80211_hw *hw)
+static void mt7615_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt7615_dev *dev = mt7615_hw_dev(hw);
 	struct mt7615_phy *phy = mt7615_hw_phy(hw);
--- a/mt7615/usb.c
+++ b/mt7615/usb.c
@@ -79,7 +79,7 @@ static void mt7663u_copy(struct mt76_dev
 	mutex_unlock(&usb->usb_ctrl_mtx);
 }
 
-static void mt7663u_stop(struct ieee80211_hw *hw)
+static void mt7663u_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt7615_phy *phy = mt7615_hw_phy(hw);
 	struct mt7615_dev *dev = hw->priv;
--- a/mt76x0/pci.c
+++ b/mt76x0/pci.c
@@ -44,7 +44,7 @@ static void mt76x0e_stop_hw(struct mt76x
 	mt76_clear(dev, MT_WPDMA_GLO_CFG, MT_WPDMA_GLO_CFG_RX_DMA_EN);
 }
 
-static void mt76x0e_stop(struct ieee80211_hw *hw)
+static void mt76x0e_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt76x02_dev *dev = hw->priv;
 
--- a/mt76x0/usb.c
+++ b/mt76x0/usb.c
@@ -77,7 +77,7 @@ static void mt76x0u_cleanup(struct mt76x
 	mt76u_queues_deinit(&dev->mt76);
 }
 
-static void mt76x0u_stop(struct ieee80211_hw *hw)
+static void mt76x0u_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt76x02_dev *dev = hw->priv;
 
--- a/mt76x2/pci_main.c
+++ b/mt76x2/pci_main.c
@@ -24,7 +24,7 @@ mt76x2_start(struct ieee80211_hw *hw)
 }
 
 static void
-mt76x2_stop(struct ieee80211_hw *hw)
+mt76x2_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt76x02_dev *dev = hw->priv;
 
--- a/mt76x2/usb_main.c
+++ b/mt76x2/usb_main.c
@@ -22,7 +22,7 @@ static int mt76x2u_start(struct ieee8021
 	return 0;
 }
 
-static void mt76x2u_stop(struct ieee80211_hw *hw)
+static void mt76x2u_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt76x02_dev *dev = hw->priv;
 
--- a/mt7915/main.c
+++ b/mt7915/main.c
@@ -108,7 +108,7 @@ static int mt7915_start(struct ieee80211
 	return ret;
 }
 
-static void mt7915_stop(struct ieee80211_hw *hw)
+static void mt7915_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt7915_dev *dev = mt7915_hw_dev(hw);
 	struct mt7915_phy *phy = mt7915_hw_phy(hw);
--- a/mt7921/main.c
+++ b/mt7921/main.c
@@ -268,7 +268,7 @@ static int mt7921_start(struct ieee80211
 	return err;
 }
 
-static void mt7921_stop(struct ieee80211_hw *hw)
+static void mt7921_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt792x_dev *dev = mt792x_hw_dev(hw);
 	int err = 0;
@@ -281,7 +281,7 @@ static void mt7921_stop(struct ieee80211
 			return;
 	}
 
-	mt792x_stop(hw);
+	mt792x_stop(hw, false);
 }
 
 static int
--- a/mt792x.h
+++ b/mt792x.h
@@ -251,7 +251,7 @@ static inline bool mt792x_dma_need_reini
 #define mt792x_mutex_release(dev)	\
 	mt76_connac_mutex_release(&(dev)->mt76, &(dev)->pm)
 
-void mt792x_stop(struct ieee80211_hw *hw);
+void mt792x_stop(struct ieee80211_hw *hw, bool suspend);
 void mt792x_pm_wake_work(struct work_struct *work);
 void mt792x_pm_power_save_work(struct work_struct *work);
 void mt792x_reset(struct mt76_dev *mdev);
@@ -368,7 +368,7 @@ void mt792xu_wr(struct mt76_dev *dev, u3
 u32 mt792xu_rmw(struct mt76_dev *dev, u32 addr, u32 mask, u32 val);
 void mt792xu_copy(struct mt76_dev *dev, u32 offset, const void *data, int len);
 void mt792xu_disconnect(struct usb_interface *usb_intf);
-void mt792xu_stop(struct ieee80211_hw *hw);
+void mt792xu_stop(struct ieee80211_hw *hw, bool suspend);
 
 static inline void
 mt792x_skb_add_usb_sdio_hdr(struct mt792x_dev *dev, struct sk_buff *skb,
--- a/mt792x_core.c
+++ b/mt792x_core.c
@@ -91,7 +91,7 @@ void mt792x_tx(struct ieee80211_hw *hw,
 }
 EXPORT_SYMBOL_GPL(mt792x_tx);
 
-void mt792x_stop(struct ieee80211_hw *hw)
+void mt792x_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt792x_dev *dev = mt792x_hw_dev(hw);
 	struct mt792x_phy *phy = mt792x_hw_phy(hw);
--- a/mt792x_usb.c
+++ b/mt792x_usb.c
@@ -285,12 +285,12 @@ int mt792xu_init_reset(struct mt792x_dev
 }
 EXPORT_SYMBOL_GPL(mt792xu_init_reset);
 
-void mt792xu_stop(struct ieee80211_hw *hw)
+void mt792xu_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt792x_dev *dev = mt792x_hw_dev(hw);
 
 	mt76u_stop_tx(&dev->mt76);
-	mt792x_stop(hw);
+	mt792x_stop(hw, false);
 }
 EXPORT_SYMBOL_GPL(mt792xu_stop);
 
--- a/mt7996/main.c
+++ b/mt7996/main.c
@@ -93,7 +93,7 @@ static int mt7996_start(struct ieee80211
 	return ret;
 }
 
-static void mt7996_stop(struct ieee80211_hw *hw)
+static void mt7996_stop(struct ieee80211_hw *hw, bool suspend)
 {
 	struct mt7996_dev *dev = mt7996_hw_dev(hw);
 	struct mt7996_phy *phy = mt7996_hw_phy(hw);
